### ISSUE TITLE
rectangle & polygon as seperate tasks (ability to redo rectangles as polys?)

### ISSUE THREAD

**Issue by dobkeratops** [ImageMonkey/imagemonkey-core#158]
would this drop out naturally from incomplete tasks, or would this make sense to seperate anyway..

it would be useful to be able to refine a rectangular annotation with a polygon.. e.g. many rectangles are placed if you're impatient or are using an inferior input device (eg laptop trackpad), but going back to something in a different frame of mind (and with a mouse or tablet) it's possible to do with greater precision..

collecting both could be useful, e.g. think of it as a form of validation if 1 person draws a box and another draws a polygon mostly covered by boxes

Of course sometimes boxes are the best way to annotate (specifically small boxes around many distant items, e.g. small people in street scenes, where the pixels are too small to draw a polygon anyway, but a box can still identify a 'tall shape' (person, post,..) versus a 'wide shape' (car,bus..)

'completeness' would have to consider both boxes and rectangles

**Comment by bbernhard** [comment]
good idea!

I think with the upcoming annotation rework feature, it should already be possible to do that (I'll create a ticket later today with more details how this new mode is intended to work)

But the whole rework feature has also a pretty severe downside: we lose labels that were defined on a per-bounding box basis. Imagine an image with three apples. Someone annotated those apples by drawing a bounding box around each apple. Now people started to add labels to those individual bounding boxes: `color: green`, `color: yellow`, `color: red`,...

Now another use comes along and reworks the annotation and replaces the bounding boxes with polygons (as they are more accurate). The problem is now, that the color labels are directly bound to the bounding boxes..and as they are now gone, we also lose the information about the apple's colors.

**Comment by bbernhard** [comment]
> maybe we could think about explicitely binding one annotation as a refinement of another or something like that

yeah, also thought about that. But as you already mentioned, it's pretty hard to find a UIX workflow that's easy and intuitive. There might be some cases where it's really just a refinement (bounding box -> polygon), whereas in other cases, someone might use the annotation rework mode to correct wrong annotations. (-> here we can and should drop any existing label refinements). 

I think we don't need a solution now, but as soon as we want to have labels on a per-annotation basis, we should probably come up with something. The thing is, you can easily run into this situation, as there is _always_ something to refine: a bounding box can be replaced with a polygon; poly points can be re-arranged to better match the objects shape...

It would be bad if we los all the information that we've collected on a per-annotation basis, if someone just re-arranges a few poly points or changes the width of a bounding box to better match the objects shape.
